File: python/2171.py
from typing import List

class Solution:
    def minimumRemoval(self, beans: List[int]) -> int:
        n = len(beans)
        beans.sort()
        preSum = [0] * n
        preSum[0] = beans[0]
        # for i, bean in enumerate(beans[1:]):
        #     preSum[i] += preSum[i-1] + bean
        # res = preSum[-1]
        s = sum(beans)
        res = s
        for i, bean in enumerate(beans):
            x = s - bean * (n - i)
            res = min(res, x)
        return res
        

File: go/2171.go
package main

import "sort"

// func sum(li []int) (ans int) {
// 	for _, v := range li {
// 		ans += v
// 	}
// 	return
// }

func minimumRemoval(beans []int) (res int64) {
	n, s := len(beans), int64(sum(beans))
	sort.Ints(beans)
	res = s
	for i, bean := range beans {
		x := s - int64(bean*(n-i))
		res = min(res, x)
	}
	return
}
